pkg/common/utils/utils_test.go

```diff
@@ -1140,3 +1211,30 @@ func TestConvert2PriorityClass(t *testing.T) {
 	assert.Assert(t, result != nil)
 	assert.Equal(t, result.PreemptionPolicy, &preemptLower)
 }
+
+func TestWaitForCondition(t *testing.T) {
+	target := false
+	eval := func() bool {
+		return target
+	}
+	tests := []struct {
+		input    bool
+		interval time.Duration
+		timeout  time.Duration
+		output   error
+	}{
+		{true, time.Duration(1) * time.Second, time.Duration(2) * time.Second, nil},
+		{false, time.Duration(1) * time.Second, time.Duration(2) * time.Second, ErrorTimeout},
+		{true, time.Duration(3) * time.Second, time.Duration(2) * time.Second, nil},
+		{false, time.Duration(3) * time.Second, time.Duration(2) * time.Second, ErrorTimeout},
+	}
+	for _, test := range tests {
+		target = test.input
+		get := WaitForCondition(eval, test.timeout, test.interval)
+		if test.output == nil {
+			assert.NilError(t, get)
+		} else {
+			assert.Equal(t, get.Error(), test.output.Error())
+		}
+	}
+}
```

Comment:
The TestWaitForCondition() takes 5~6 secs (the second and forth test cases) , I guess it's accepted. But  I'm also thinking about whether we could use a smaller `interval` and `timout`, like 100 Millisecond and 200 Millisecond?

The only concern is I'm not sure whehter our CI env's goroutine could got CPU in time.
@chia7712 WDYT?

Reply:
Or, maybe we could removie this test? I'm not expecting any code change to this function in the future.

Reply:
I guess that test case is another sort of "consistent result"  https://github.com/apache/yunikorn-core/blob/21303191a9c9ee791de3371ddca973df51b7a755/pkg/common/utils_test.go#L239

Hence, I'm fine to keep this test case :)

Reply:
OK, let's keep the test cases consistent with core.